confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java

```diff
@@ -496,19 +497,23 @@ public String getRegionStorageGroup(TConsensusGroupId regionId) {
 
   /** Called by {@link PartitionManager#regionCleaner} Delete RegionGroups periodically. */
   public void clearDeletedRegions() {
-    if (getConsensusManager().isLeader()) {
-      final Set<TRegionReplicaSet> deletedRegionSet = partitionInfo.getDeletedRegionSet();
-      if (!deletedRegionSet.isEmpty()) {
-        LOGGER.info(
-            "DELETE REGIONS {} START",
-            deletedRegionSet.stream()
-                .map(TRegionReplicaSet::getRegionId)
-                .collect(Collectors.toList()));
-        deletedRegionSet.forEach(
-            regionReplicaSet -> removeRegionGroupCache(regionReplicaSet.regionId));
-        SyncDataNodeClientPool.getInstance().deleteRegions(deletedRegionSet);
-      }
-    }
+    Optional.ofNullable(getConsensusManager())
+        .ifPresent(
```

Comment:
We'd better add annotation about why should we must use Optional.ofNullable.ifPresent here.

Reply:
Done